loadbalancer_interface/provides.py

```diff
@@ -105,6 +119,11 @@ def removed_requests(self):
 
     def send_response(self, request):
         """Send a specific request's response."""
+        if not self.unit.is_leader():
+            # This unit is a follower which cannot write to
+            # relation.data[self.app]
+            return
```

Comment:
I would consider logging a warning if a follower tries to send a response when it can't